### PR TITLE
Add support for DynamoDB events to aws_cloudtrail

### LINKED ISSUE #18394: Add support for DynamoDB events to aws_cloudtrail
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS just added support for DynamoDB events to CloudTrail https://aws.amazon.com/about-aws/whats-new/2021/03/aws-cloudtrail-adds-logging-of-data-events-for-amazon-dynamoDB/ would be great to be able to configure this with terrraform.

### New or Affected Resource(s)

* aws_cloudtrail

### PR DESCRIPTION
**Pull request by bishtawi**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #18394
Superceded by #19559

Sorry, I am unable to run the acceptance tests as I dont have an available AWS account that I can run tests on. I was able to successfully execute `make build` and `make test` if that is worth anything. The change is pretty minimal though.

Should I update the `testAccAWSCloudTrail_event_selector` test? I wasnt sure if that was necessary since this PR is just adding a string to an existing validation function.

### CODE REVIEW

File: website/docs/r/cloudtrail.html.markdown [hunk 1]
```diff
@@ -185,8 +185,8 @@ For **event_selector** the following attributes are supported.
 #### Data Resource Arguments
 For **data_resource** the following attributes are supported.
 
-* `type` (Required) - The resource type in which you want to log data events. You can specify only the following value: "AWS::S3::Object", "AWS::Lambda::Function"
-* `values` (Required) - A list of ARN for the specified S3 buckets and object prefixes..
+* `type` (Required) - The resource type in which you want to log data events. You can specify only the following value: "AWS::S3::Object", "AWS::Lambda::Function" and "AWS::DynamoDB::Table".
+* `values` (Required) - A list of ARN for the specified S3 buckets and object prefixes.
```
Comment (ryanbreed-clearme) [inline website/docs/r/cloudtrail.html.markdown:189]: * `values` (Required) - A list of ARN strings or partial ARN strings to specify selectors for data audit events over data resources. ARN list is specific to single-valued `type`
  * `arn:aws:s3:::<bucket_name>/` - all objects in bucket
  * `arn:aws:s3:::<bucket_name>/key` specific key(s) 
  * `arn:aws:lambda` - all lambda events within account
  * `arn:aws:lambda:us-west-2:111111111111:function:helloworld` - events for exact arn
  * `arn:aws:dynamodb` - all DDB events for all tables within account

Comment (bishtawi) [reply website/docs/r/cloudtrail.html.markdown]: Updated

**Comment by antontumasov-plutoflume** [comment]
Hey - any ETA when this gets released? We're really blocked by this ATM and would appreciate this coming in the next few releases.

**Comment by grahamc** [comment]
> Hey - any ETA when this gets released? We're really blocked by this ATM and would appreciate this coming in the next few releases.

For what it is worth, it can take a long time to get patches merged. If this is blocking you, you might consider patching and building the provider yourself. I know this isn't ideal, but I've found it to be fairly common to need to do that.

**Comment by YakDriver** [comment]
@bishtawi Thank you for this PR! I am looking into this now. Two items I notice that would speed up the merge of this PR are:

1. Include a new acceptance test that tests the new functionality
2. Use a fork branch that is not called "main" (convention is to use `f-` for a feature branch with a short description; for example, this branch could be called `f-cloudtrail-dynamodb-events`). Using "main" makes it difficult to work with because the actual main on the main repository is called main.

However, since this PR is quite popular, I will work on finishing this for you. No guarantees on a timeframe.

**Comment by YakDriver** [comment]
@bishtawi Thank you for your work! I have cherry picked your commits to a new branch (see #19559). You will still get credit for them. This makes working with this PR easier.

**Pull request closed without merging**